modules/components/tag-input/tag-input.ts

```diff
@@ -509,11 +509,20 @@ export class TagInputComponent extends TagInputAccessor implements OnInit, After
      * @name appendTag
      * @param tag {TagModel}
      */
-    public appendTag = (tag: TagModel): void => {
-        this.items = [
-            ...this.items,
-            this.modelAsStrings ? tag[this.identifyBy] : tag
-        ];
+    public appendTag = (tag: TagModel, index = this.items.length): void => {
+        const items = [];
+
+        for (let i = 0; i <= this.items.length; i++) {
+            if (i < index) {
+                items.push(this.items[i]);
+            } else if (i === index) {
+                items.push(this.modelAsStrings ? tag[this.identifyBy] : tag);
+            } else {
+                items.push(this.items[i - 1]);
+            }
+        }
```

Comment:
The original version would only insert the new tag at the end of the array, whereas in the case of drag&drop, the tag can be placed anywhere in the array.

Since this function is now used for drag&drop as well, it has to support an index.